graph/graph.go

```diff
@@ -93,8 +99,21 @@ func (g *Graph) GetParent(id string) interface{} {
 			break
 		}
 	}
+	return parentID
+}
 
-	return g.Get(parentID)
+// IsNibling checks to see if second is the child of a sibling of the first.
+func (g *Graph) IsNibling(fst, snd string) bool {
+	if snd == "." || snd == "" {
+		return false
+	}
+	if fst == g.GetParentID(snd) {
+		return false
+	}
+	if AreSiblingIDs(fst, snd) {
```

Comment:
this is still doing the calculation based on IDs. Could you change it to use *only* the metadata in the graph, or would that be too much?